.dotnet/src/Generated/Chat.cs

```diff
@@ -151,33 +147,31 @@ public virtual Result CreateChatCompletion(RequestBody content, RequestOptions c
             }
         }
 
-        internal PipelineMessage CreateCreateChatCompletionRequest(RequestBody content, RequestOptions context)
+        internal PipelineMessage CreateCreateChatCompletionRequest(BinaryContent content, RequestOptions options)
         {
-            var message = _pipeline.CreateMessage(context, ResponseErrorClassifier200);
-            var request = message.Request;
-            request.SetMethod("POST");
-            var uri = new RequestUri();
-            uri.Reset(_endpoint);
-            uri.AppendPath("/chat/completions", false);
-            request.Uri = uri.ToUri();
-            request.SetHeaderValue("Accept", "application/json");
-            request.SetHeaderValue("Content-Type", "application/json");
+            PipelineMessage message = _pipeline.CreateMessage();
+            message.ResponseClassifier = ResponseErrorClassifier200;
+
+            PipelineRequest request = message.Request;
+            request.Method = "POST";
+
+            UriBuilder uriBuilder = new(_endpoint.ToString());
+            StringBuilder path = new();
+            path.Append("/chat/completions");
+            uriBuilder.Path += path.ToString();
+            request.Uri = uriBuilder.Uri;
+
+            request.Headers.Set("Accept", "application/json");
+            request.Headers.Set("Content-Type", "application/json");
+
             request.Content = content;
-            return message;
-        }
 
-        private static RequestOptions DefaultRequestContext = new RequestOptions();
-        internal static RequestOptions FromCancellationToken(CancellationToken cancellationToken = default)
-        {
-            if (!cancellationToken.CanBeCanceled)
-            {
-                return DefaultRequestContext;
-            }
+            message.Apply(options);
```

Comment:
Add call to message.Apply(options) -- important that this goes after headers are added to request.